specification/subscription/resource-manager/Microsoft.Subscription/preview/2021-10-01-preview/cancelEligibility.json

```diff
@@ -0,0 +1,180 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/CancelEligibilityResponse"
+            }
+          },
+          "default": {
+            "description": "Describes the error if the operation is not successful.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponseBody"
+            }
+          },
+          "x-ms-pageable": {
+            "nextLinkName": "nextLink"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CancelEligibilityResponse": {
+      "type": "object",
+      "properties": {
+        "isEligibleToCancel": {
+          "type": "boolean",
+          "description": "Is eligible to cancel"
+        }
+      },
+      "description": "Cancel Eligibility Response",
+      "required": [
+        "isEligibleToCancel"
+      ]
+    },
+    "ErrorResponseBody": {
+      "type": "object",
+      "properties": {
+        "error": {
+          "$ref": "#/definitions/Reasons"
+        }
+      },
+      "required": [
+        "error"
+      ]
+    },
+    "Reason": {
+      "type": "object",
+      "properties": {
+        "SeverityLevel": {
+          "type": "string"
+        },
+        "ErrorCode": {
+          "type": "string"
+        },
+        "ErrorMessage": {
+          "type": "string"
+        },
+        "ResourceLockOwners": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "x-typespec-name": "string[]"
+        },
+        "Resources": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "x-typespec-name": "string[]"
+        }
+      },
+      "required": [
+        "SeverityLevel",
+        "ErrorCode",
+        "ErrorMessage"
+      ]
+    },
+    "Reasons": {
+      "type": "object",
+      "properties": {
+        "reasons": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Reason"
+          },
+          "x-ms-identifiers": [],
+          "x-typespec-name": "Reason[]"
+        },
+        "nextLink": {
+          "description": "The link (url) to the next page of results.",
+          "type": "string",
+          "readOnly": true
+        }
+      },
+      "required": [
+        "reasons",
+        "nextLink"
+      ]
+    }
+  },
+  "parameters": {
+    "apiVersionParameter": {
+      "name": "api-version",
+      "in": "query",
+      "required": true,
+      "type": "string",
+      "description": "Version of the API to be used with the client request. Current version is 2021-10-01-preview"
+    },
+    "subscriptionIdParameter": {
+      "name": "subscriptionId",
+      "x-ms-parameter-location": "method",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "Subscription Id."
+    }
```

Comment:
Instead of defining these yourself, please use $ref to the ones in common-types.



[https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/types.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5/types.json)